tooling/vdu-prechecks/vdu_precheck.py

```diff
@@ -0,0 +1,302 @@
+                print(f"want: {key}: {reference_bios[vendor][key]}")
+                print(f"have: {key}: {bios[key]}")
+            if reference_bios[vendor][key] != bios[key]:
+                print("BIOS mismatch found")
+                errors = True
+                print(f"want: {key}: {reference_bios[vendor][key]}")
+                print(f"have: {key}: {bios[key]}")
+        else:
+            print(f"key not found: {key}")
+            errors = True
+    if not errors:
+        print("BIOS: ok")
+
+
+def get_nics(sep, vendor):
+    '''
+    displays the details of any supported Ethernet devices discovered on the system
+    '''
+
+    nics = []
+
+    if vendor == "HPE":
+        url = f"https://{args.ip}{sep}/BaseNetworkAdapters"
+    elif vendor == "Dell Inc.":
+        url = f"https://{args.ip}{sep}/NetworkAdapters"
+    else:
+        print(f"unsupported server vendor: {vendor}")
+        return
+
+    members = get_url(url)['Members']
+    nics = []
+    for nic in members:
+        url = f"https://{args.ip}{nic['@odata.id']}"
+        nic_data = get_url(url)
+        nic = parse_nic(nic_data, vendor)
+        if nic:
+            if isinstance(nic, list):
+                for port in nic:
+                    nics.append(port)
+            else:
+                nics.append(nic)
+    if len(nics) == 0:
+        print("no supported nics found")
+    else:
+        print("NICs:")
+        dpp.pprint(nics)
+
+def parse_nic(nic_data, vendor):
+    '''
+    parse nic data based on vendor
+    '''
+
+    nics = []
+
+    if vendor == "HPE":
+        nic_model = nic_data['Name']
+        if supported_nic(nic_model):
+            nic_slot = nic_data['Location']
+            macs = []
+            for port in nic_data['PhysicalPorts']:
+                macs.append(port['MacAddress'])
+            return {
+              'model': nic_model,
+              'slot': nic_slot,
+              'macs': macs
+            }
+
+    if vendor == "Dell Inc.":
+        nic_model = nic_data['Model']
+        if supported_nic(nic_model):
+            #print(nic_data)
+            ports = nic_data['NetworkPorts']['@odata.id']
+            url = f"https://{args.ip}{ports}"
+            members = get_url(url)['Members']
+            for member in members:
+                url = f"https://{args.ip}{member['@odata.id']}"
+                port_data = get_url(url)
+                nic_slot = port_data['Id']
+                macs = []
+                for mac in port_data['AssociatedNetworkAddresses']:
+                    macs.append(mac)
+                nics.append({
+                    'model': nic_model,
+                    'slot': nic_slot,
+                    'macs': macs
+                })
+            return nics
+
+    return None
+
+def supported_nic(model):
+    '''
+    test if a given NIC is supported
+    '''
+
+    for supported in supported_nics:
+        if supported in model:
+            return True
+
+    return False
```

Comment:
you can make it a oneliner with
return len([supported in model for supported in supported_nics]) > 0